mri/reconstruct/fourier.py

```diff
@@ -521,3 +521,114 @@ def adj_op(self, coeffs):
             inverse discrete Fourier transform of the input coefficients.
         """
         return self.implementation.adj_op(coeffs)
+
+
+class Stacked3D(FourierBase):
+    """"  3-D non uniform Fast Fourier Transform class,
+        fast implementation for Stacked samples. Note that the kspace locations
+        must be in the form of a stack along z, with same locations in
+        each plane
+    Attributes
+    ----------
+    samples: np.ndarray
+        the mask samples in the Fourier domain.
+    shape: tuple of int
+        shape of the image (necessarly a square/cubic matrix).
+    implementation: string, 'cpu', 'cuda' or 'opencl' default 'cpu'
+        string indicating which implemenmtation of Noncartesian FFT
+        must be carried out
+    """
+
+    def __init__(self, kspace_loc, shape, implementation='cpu', n_coils=1):
+        """ Init function for Stacked3D class
+        Parameters
+        ----------
+        kspace_loc: np.ndarray
+            the position of the samples in the k-space
+        shape: tuple of int
+            shape of the image stack in 3D. (N x N x Nz)
+        implementation: string, 'cpu', 'cuda' or 'opencl' default 'cpu'
+            string indicating which implemenmtation of Noncartesian FFT
+            must be carried out. Please refer to Documentation of
+            NoncartesianFFT
+        """
+        self.num_slices = shape[2]
+        (plane_samples, self.z_samples, self.stack_len,
+         self.acq_num_slices, self.sort_pos) = get_stacks_fourier(kspace_loc)
+        self.FT = NonCartesianFFT(samples=plane_samples, shape=shape[0:2],
+                                  implementation=implementation)
+        self.nb_coils = n_coils
```

Comment:
Why not the same name?

Reply:
Legacy, Will keep it same here and will fix in other cases going forward